### PR TITLE
[SYCL][Fusion] Take auxiliary resources from fused command groups

### PR DESCRIPTION
**Pull request by victor-eds**
Reductions use auxiliary resources to handle reduction temporal
buffers. Assign those auxiliary resources from each fused command
group to the placeholder fusion event. This event will not be marked
as completed after the fused reductions finish execution either if
fusion is completed, cancelled or aborted.

Test is updated to check every algorithm that is selected
automatically by `sycl::reduction`, i.e., every supported
algorithm. We also cover both cases (fusion taking and not taking
place).

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

**Comment by victor-eds** [comment]
Depends on #12556 (for test)

**Review by bso-intel — APPROVED** [review]

File: sycl/test-e2e/KernelFusion/Reduction/group_reduce_and_last_wg_detection.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: %{build} -fsycl-embed-ir -o %t.out
+// RUN: %{run} %t.out
+// UNSUPPORTED: hip || cuda
```
Comment (sommerlukas) [inline sycl/test-e2e/KernelFusion/Reduction/group_reduce_and_last_wg_detection.cpp:3]: Add a comment (`COM:`) why this is unsupported.

**Review by sommerlukas — APPROVED** [review]
Just one nit.

File: sycl/test-e2e/KernelFusion/Reduction/group_reduce_and_last_wg_detection.cpp (hunk 1 above)
Comment (victor-eds) [reply sycl/test-e2e/KernelFusion/Reduction/group_reduce_and_last_wg_detection.cpp:3]: Done

**Comment by sommerlukas** [comment]
Related failure of post-commit: 
```
Timed Out Tests (1):
  SYCL :: KernelFusion/Reduction/group_reduce_and_last_wg_detection.cpp
```

https://github.com/intel/llvm/actions/runs/7817537968/job/21327112200


https://github.com/intel/llvm/pull/12621 should fix this issue.

**Pull request merged**